File: renders/facilities/Park.cpp
/* NIM/Nama  : 13515AAA / AAA AAA   */
/* File      : Park.cpp             */

#include <iostream>
#include "Park.h"
using namespace std;

/***** PARK *****/
Park::Park()  : id('*'), color(WHITE)
{}

Park::Park(const Park&)  : id('*'), color(WHITE)
{}

Park::~Park()
{}

Park& Park::operator=(const Park&)
{}

char render() const
{
	return id;
}

Park* Park::clone() const {
  return new Park(*this);
}

File: Park.java
/**
 * @author NIM / Nama : 13515087 / Audry Nyonata
 *  Nama file : Park.java
 */

import java.util.*;
import java.lang.*;

public class Park extends Facility{
   /**
   * Constructor
   */
  public Park(){
    super();
  }
  /**
   * Mengembalikan karakter untuk ditampilkan ke layar
   * @return id bertipe char
   */
  public char render(){
    return '*';
  }
  /**
   * Mengembalikan warna untuk ditampilkan ke layar
   * @return color warna renderable
   */
  public String renderWithColor(){
    return ANSI_WHITE + render() + ANSI_RESET;
  }
}
